src/app/beer_garden/app.py

```diff
@@ -287,8 +281,20 @@ def _setup_scheduler():
 
     @staticmethod
     def _publish_update(event: Events):
-        garden = beer_garden.garden.get_garden(beer_garden.config.get("garden.name"))
-        garden.status = "RUNNING" if event == Events.GARDEN_STARTED else "STOPPED"
+        # DO NOT DO THIS
+        # garden = beer_garden.garden.get_garden(beer_garden.config.get("garden.name"))
+
+        # INSTEAD, THIS
+        # Want to have most current info when publishing
+        known_systems = list(beer_garden.router.garden_lookup.keys())
+        known_namespaces = beer_garden.namespace.get_namespaces()
+
+        garden = Garden(
+            name=beer_garden.config.get("garden.name"),
+            status="RUNNING" if event == Events.GARDEN_STARTED else "STOPPED",
+            systems=known_systems,
+            namespaces=known_namespaces,
+        )
 
         publish(Event(name=event.name, payload_type="Garden", payload=garden))
```

Comment:
This could cause issues. When I was first looking at this, when you add all known systems and namespaces to the garden, then publish it, the `Router` class would take that are the new routing information. So it would never try to route to a downstream garden, it would always try to publish it locally.

To verify that this is not a problem, please give this a test:
1) Deploy Parent Garden
2) Deploy Child Garden w/ plugins
3) Task Child Plugin
4) Restart Parent Garden
5) Task Child Plugin

Reply:
This works as expected for me.

Reply:
:+1: